modules/vault-cluster/main.tf

```diff
@@ -8,6 +8,41 @@ terraform {
   required_version = ">= 0.10.3"
 }
 
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATES A SERVICE ACCOUNT TO OPERATE THE VAULT CLUSTER
+# The default project service account will be used if create_service_account
+# is set to false and no service_account_email is provided.
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "google_service_account" "vault_cluster_admin" {
+  count        = "${var.create_service_account}"
+  account_id   = "${var.cluster_name}-admin-sa"
+  display_name = "Vault Server Admin"
+  project      = "${var.gcp_project_id}"
+}
+
+# Create a service account key
+resource "google_service_account_key" "vault" {
+  count              = "${var.create_service_account}"
+  service_account_id = "${google_service_account.vault_cluster_admin.name}"
```

Comment:
Since `google_service_account.vault_cluster_admin` uses `count`, you can't reference it this way. Instead, you have to use an ugly workaround: `${element(list(google_service_account.vault_cluster_admin.*.name, ""), 0)}`. Same with all of these resources here that use `count`, any cross-references between them, and any references to them in `outputs.tf`.

Reply:
This one specifically won't error, because it uses the same variable for the count, so this resource doesn't get created either and doesn't hit this limitation

Reply:
I'm reasonably confident Terraform will give you at least a warning just for referencing a resource that way that has a `count` param. And that warning could well become an error in the future, so it's worth changing to the full syntax IMO.